File: src/Exporter.NX.Translate/IdAllocator.cs
// SPDX-License-Identifier: GPL-2.0-only
namespace Oblikovati.Exporter.NX.Translate
{
    /// <summary>
    /// Hands out monotonically increasing document-local ids. Sketches, points and
    /// entities share one id space (matching the Go codec, where a sketch's id precedes
    /// its points' and entities' ids), so one allocator threads through the document.
    /// </summary>
    public sealed class IdAllocator
    {
        private int _next;

        public IdAllocator(int start = 1)
        {
            _next = start;
        }

        public int Next() => _next++;
    }
}

File: python/oblikovati_exporter_nx/translate/id_allocator.py
# SPDX-License-Identifier: GPL-2.0-only
"""Hands out monotonically increasing document-local ids.

Sketches, points and entities share one id space (matching the Go codec, where a
sketch's id precedes its points' and entities' ids), so one allocator threads through
the document.
"""
from __future__ import annotations


class IdAllocator:
    def __init__(self, start: int = 1) -> None:
        self._next = start

    def next(self) -> int:
        value = self._next
        self._next += 1
        return value
